agent/engine/docker_container_engine.go

```diff
@@ -371,6 +336,50 @@ func (dg *dockerGoClient) pullImage(image string, authData *api.RegistryAuthenti
 	return nil
 }
 
+func (dg *dockerGoClient) filterDebugOutput(pullDebugOut *io.PipeReader, pullBegan chan bool, image string) {
+	// pullBeganOnce ensures we only indicate it began once (since our channel will only be read 0 or 1 times)
+	pullBeganOnce := sync.Once{}
+
+	reader := bufio.NewReader(pullDebugOut)
+	var line string
+	var pullErr error
+	var statusDisplayed time.Time
+	for pullErr == nil {
```

Comment:
can this be just `for {`? We already do the `if pullErr != nil { break }` within the method anyway. So, I think that should be fine.

Reply:
You are right, for condition here is not doing anything. Removed.